test/app/model/BoardFixture.ts

```diff
@@ -0,0 +1,15 @@
+import { Board } from '../../../src/app/model/Board';
+import { Square } from '../../../src/app/model/Types';
+import { Piece } from '../../../src/app/model/Piece';
+
+export const emptyBoard: Board = { onPositionPiece: (square: Square) => null };
+export type SquareWithPiece = { [key: string]: Piece };
+
+export const boardWithPieces: (squaresWithPiece: SquareWithPiece) => Board = (squaresWithPiece) => {
+  return {
+    onPositionPiece: (square: Square) => {
+      const pieceOnSquare = squaresWithPiece[`${square.column}${square.row}`];
+      return pieceOnSquare ? pieceOnSquare : null;
+    },
+  };
+};
```

Comment:
Czy to powinno być umieszczone w katalogu "model" czy może w folderze/podfolderze o jakiejś nazwie np "units"?

Reply:
Na razie bym nie przesadzał, póki nie mamy tego dużo. Chyba jest OK. 
Widać, że to nie plik testowy po braku .spec.ts + Fixture to pattern znany na wyróżnienie takich przygotowanych obiektów do testów.

Reply:
Ok, bo pomyslałem, aby może od razu powrzucać (np. do "Fixtures").  
Bo w ostatnim projekcie zrobił się mały śmietnik z powodu nie umieszczania plików tam gdzie powinny się znaleźć.
A takich plików może być więcej, jeśli nas fantazja poniesie

Reply:
to jaka decyzja? zostajemy przy tym `null` czy zmiana banana na `undefined` ? ^^

Reply:
@tomdworniczak jak bym został przy null, bo wtedy przynajmniej wiemy, że działa, jak Ci pójdzie null ... a otrzymując undefined, może być dobrze, a może być tak, że faktycznie nie jest coś zdefiniowane, ale to jest tylko moje zdanie (możliwe, że nie będzie takiej sytuacji, tak głęboko w to nie wchodziłem :-) )

Reply:
To co opisałem tutaj:
![image](https://user-images.githubusercontent.com/18637116/106124759-bca85a00-615b-11eb-8ec2-4273cb07821a.png)

Reply:
No i reszta dyskusji :) 

Tzn... możecie używać null i kiedyś zdecydować, że trzeba przejść na undefined, albo zrobić to od razu. 
Jak chcecie się przekonać ile jest walenia z nullem.

Porównaj @Szambelan np. do Javy. Odpowiednikiem z Javy nulla jest w JS undefined. Nie potrzebne nam rozróżnienie czy coś jest niezdefiniowane albo ktoś przypisał tam undefined. 
Tzn. zadaj sobie pytanie - czy potrafisz wyobrazić sobie jakąś sytuację, żeby Twoja reakcja była inna na undefined niż na null? A kod przy nullu się komplikuje.

Reply:
właśnie to co napisał Mateusz mnie przekonuje bardziej do `undefined`, chyba cała ta dyskusja opiera się na tym, że trochę opacznie postrzegamy `undefined`  :P

Reply:
no dobra to zmieńmy na undefined :-)

Reply:
@Szambelan Zmienilem na undefined i merguje. Wez to pod uwage przy kolejnym tasku :)